### ISSUE TITLE
Too many open files (EMFILE) when doing `Pkg.update()` on v0.3.-

### ISSUE THREAD

**Issue by jakebolewski** [JuliaLang/julia#8601]
I installed every registered Julia pakage to test against.  On OS X I'm seeing this error (it makes it down to the F's before dying).  Trying to call `Pkg.update()` again segfaults.

```shell
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "help()" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.3.1 (2014-09-21 21:30 UTC)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |  x86_64-apple-darwin13.4.0

julia> Pkg.update()
INFO: Updating METADATA...
ERROR: could not spawn setenv(`git rev-parse --git-dir`; dir="/Users/jacobbolewski/.julia/v0.3/.cache/FunctionalCollections"): too many open files (EMFILE)
 in wait at task.jl:51
 in sync_end at /usr/local/Cellar/julia/0.3.1/lib/julia/sys.dylib
 in update at /usr/local/Cellar/julia/0.3.1/lib/julia/sys.dylib
 in anonymous at pkg/dir.jl:28
 in cd at /usr/local/Cellar/julia/0.3.1/lib/julia/sys.dylib
 in __cd#227__ at /usr/local/Cellar/julia/0.3.1/lib/julia/sys.dylib
 in update at /usr/local/Cellar/julia/0.3.1/lib/julia/sys.dylib (repeats 2 times)

julia> Pkg.update()
INFO: Updating METADATA...
Assertion failed: (loop->watchers[w->fd] == w), function uv__io_stop, file src/unix/core.c, line 764.

signal (6): Abort trap: 6
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
[1]    93827 abort      julia
```

**Comment by jiahao** [comment]
bump

**Comment by tkelman** [comment]
Instead of just using stock `@sync` and `@async` here https://github.com/JuliaLang/julia/blob/ef062117bab19959a6905b1bb6d7963b42dddea4/base/pkg/entry.jl#L255-L278 I think it might make sense to limit the number of simultaneous instances of git we spawn to `CPU_CORES` ?

**Comment by jiahao** [comment]
One hack to increase the OSX kernel limits from 10,240 (!), e.g. with

```
sysctl -w kern.maxfiles=20480 (or whatever number you choose)
sysctl -w kern.maxfilesperproc=18000 (or whatever number you choose)
```

Ref: http://superuser.com/questions/433746/is-there-a-fix-for-the-too-many-open-files-in-system-error-on-os-x-10-7-1

**Comment by jakebolewski** [comment]
Closed by #11152
